corehq/apps/accounting/tasks.py

```diff
@@ -810,6 +812,44 @@ def calculate_users_in_all_domains(today=None):
     auto_deactivate_mobile_workers.delay()
 
 
+def count_form_submitting_mobile_workers(domain, start, end):
+    """
+    Returns the count of mobile workers who have submitted a form in the time specified.
+    """
+    form_counts_by_worker = (
+        FormES(for_export=True)
+        .domain(domain)
+        .submitted(gte=start, lt=end)
+        .user_type('mobile')
+        .user_aggregation()
+        .size(0)
+        .run()
+        .aggregations.user.normalized_buckets
+    )
+    return len(form_counts_by_worker)
+
+
+@periodic_task(run_every=crontab(hour=1, minute=0, day_of_month='1'), acks_late=True)
+def calculate_form_submitting_mobile_workers_in_all_domains(today=None):
```

Comment:
We did a sanity-check on this, pulling the most form-dense domain on staging and determining how long a similar elasticsearch query would take to aggregate by users -- it completed in milliseconds. While production does have a lot of domains, and we could probably optimize this query by batching several domains per elasticsearch call, I think this query is safe as is.

Reply:
So, exactly as written this runs in about 35 seconds on staging. I'm betting iterating through all domains and writing a `FormSubmittingMobileWorkerHistory` is the bottleneck here, since we found that the ES query for any given domain is near-instantaneous. If production has around an order of magnitude more data than staging, that puts this task at roughly 5 minutes to run.

The other option, I think, is to only write the `FormSubmittingMobileWorkerHistory` for domains on a subscription that uses that feature, by prefiltering the list of domains. Something like:
```python
    form_submitting_feature_rates = FeatureRate.objects.filter(
        feature__feature_type=FeatureType.FORM_SUBMITTING_MOBILE_WORKER
    )
    relevant_domains = Subscription.visible_objects.filter(
        plan_version__feature_rates__in=form_submitting_feature_rates
    ).values_list('subscriber__domain').distinct()

    for domain in relevant_domains:
```